File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/TutunKullanimiVeriSetiFormViewModel.cs
//$00410AD8
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class TutunKullanimiVeriSetiServiceController
    {
        partial void PreScript_TutunKullanimiVeriSetiForm(TutunKullanimiVeriSetiFormViewModel viewModel, TutunKullanimiVeriSeti tutunKullanimiVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
               
                
                viewModel._TutunKullanimiVeriSeti.EpisodeAction = episodeAction;
                viewModel._TutunKullanimiVeriSeti.SubEpisode = episodeAction.SubEpisode;
                

            }

            ContextToViewModel(viewModel, objectContext);
        }
    }
}

namespace Core.Models
{
    public partial class TutunKullanimiVeriSetiFormViewModel: BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._TutunKullanimiVeriSeti);

            if (this.TutunKullanimiBagimOldUrunGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.TutunKullanimiBagimOldUrunGridList)
                {
                    var tutunKullanimiBagimOldUrun = (TutunKullanimiBagimOldUrun)objectContext.AddObject(item);
                    tutunKullanimiBagimOldUrun.TutunKullanimiVeriSeti = this._TutunKullanimiVeriSeti;
                }
            }

            if (this.TutunKullanimiTedaviSekliGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.TutunKullanimiTedaviSekliGridList)
                {
                    var tutunKullanimiTedaviSekli = (TutunKullanimiTedaviSekli)objectContext.AddObject(item);
                    tutunKullanimiTedaviSekli.TutunKullanimiVeriSeti = this._TutunKullanimiVeriSeti;
                }
            }

            if (this.TutunKullanimiTedaviSonucuGridList != null)
            {
                foreach (var item in this.TutunKullanimiTedaviSonucuGridList)
                {
                    var tutunKullanimiTedaviSonucu = (TutunKullanimiTedaviSonucu)objectContext.AddObject(item);
                    tutunKullanimiTedaviSonucu.TutunKullanimiVeriSeti = this._TutunKullanimiVeriSeti;
                }
            }

           



            var tutunKullanimiVeriSeti = (TutunKullanimiVeriSeti)objectContext.GetLoadedObject(this._TutunKullanimiVeriSeti.ObjectID);


            new SendToENabiz(objectContext, tutunKullanimiVeriSeti.SubEpisode, tutunKullanimiVeriSeti.ObjectID, tutunKullanimiVeriSeti.ObjectDef.Name, "248", Common.RecTime());
        }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/TutunKullanimiVeriSetiFormViewModel.ts
//$75582980
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { TutunKullanimiVeriSeti } from "NebulaClient/Model/AtlasClientModel";
import { TutunKullanimiTedaviSonucu } from "NebulaClient/Model/AtlasClientModel";
import { TutunKullanimiTedaviSekli } from "NebulaClient/Model/AtlasClientModel";
import { TutunKullanimiBagimOldUrun } from "NebulaClient/Model/AtlasClientModel";
import { SKRSTutunDumaninaMaruzKalmaPasificicilik } from "NebulaClient/Model/AtlasClientModel";
import { SKRSSigaraKullanimi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSTutunTedaviSonucu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSTedaviSekli } from "NebulaClient/Model/AtlasClientModel";
import { SKRSBagimliOlduguUrun } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';


export class TutunKullanimiVeriSetiFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.TutunKullanimiVeriSetiFormViewModel, Core';
    @Type(() => TutunKullanimiVeriSeti)
    public _TutunKullanimiVeriSeti: TutunKullanimiVeriSeti = new TutunKullanimiVeriSeti();
    @Type(() => TutunKullanimiTedaviSonucu)
    public TutunKullanimiTedaviSonucuGridList: Array<TutunKullanimiTedaviSonucu> = new Array<TutunKullanimiTedaviSonucu>();
    @Type(() => TutunKullanimiTedaviSekli)
    public TutunKullanimiTedaviSekliGridList: Array<TutunKullanimiTedaviSekli> = new Array<TutunKullanimiTedaviSekli>();
    @Type(() => TutunKullanimiBagimOldUrun)
    public TutunKullanimiBagimOldUrunGridList: Array<TutunKullanimiBagimOldUrun> = new Array<TutunKullanimiBagimOldUrun>();
    @Type(() => SKRSTutunDumaninaMaruzKalmaPasificicilik)
    public SKRSTutunDumaninaMaruzKalmaPasificiciliks: Array<SKRSTutunDumaninaMaruzKalmaPasificicilik> = new Array<SKRSTutunDumaninaMaruzKalmaPasificicilik>();
    @Type(() => SKRSSigaraKullanimi)
    public SKRSSigaraKullanimis: Array<SKRSSigaraKullanimi> = new Array<SKRSSigaraKullanimi>();
    @Type(() => SKRSTutunTedaviSonucu)
    public SKRSTutunTedaviSonucus: Array<SKRSTutunTedaviSonucu> = new Array<SKRSTutunTedaviSonucu>();
    @Type(() => SKRSTedaviSekli)
    public SKRSTedaviSeklis: Array<SKRSTedaviSekli> = new Array<SKRSTedaviSekli>();
    @Type(() => SKRSBagimliOlduguUrun)
    public SKRSBagimliOlduguUruns: Array<SKRSBagimliOlduguUrun> = new Array<SKRSBagimliOlduguUrun>();
}
